### PR TITLE
add html5video-rules

### PR DESCRIPTION
**Pull request by Xuanwo**
Hi, @NicolasSiver I test these rules on my own nodebb in ACP, but may have some problem in json format, please check that. I insert a new rule list and hope it can appear besides the default one.

And due to you wonderful work, I think I don't need to maintain my [plugin](https://github.com/Xuanwo/nodebb-plugin-html5videoplayer) any more.

### CODE REVIEW

File: data/html5video-rules.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+    "rules": [
+        {
+            "name": "MOV",
+            "displayName" : "MOV",
+            "icon": "",
+            "regex": "<a href="(https|http)?:?(.*\.mov)".*>.*<\/a>",
```
Comment (NicolasSiver) [inline data/html5video-rules.json]: Anchor tag isn't optional. As a reference you can use default rules

File: data/html5video-rules.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+    "rules": [
```
Comment (NicolasSiver) [inline data/html5video-rules.json:2]: Is it possible to have single rule for all video formats?
As I can see, this change is about single concern - represent direct video url as a video tag.

File: data/html5video-rules.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            "icon": "",
+            "regex": "<a href="(https|http)?:?(.*\.mov)".*>.*<\/a>",
+            "replacement": "<div class='embed-container' data-src="$1" data-type="video/mp4" data-codec="avc1.42E01E, mp4a.40.2" ><br><video width="100%" height="auto" preload controls><source src="$1" type='video/mp4; codecs="avc1.42E01E, mp4a.40.2"' /></video></div>"
+        },
+        {
+            "name": "MP4",
+            "displayName" : "MP4",
+            "icon": "",
+            "regex": "<a href="(https|http)?:?(.*\.mp4)".*>.*<\/a>",
+            "replacement": "<div class="embed-container"  data-src="$1" data-type="video/mp4" data-codec="avc1.42E01E, mp4a.40.2" ><br><video width="100%" height="auto" preload controls><source src="$1" type='video/mp4; codecs="avc1.42E01E, mp4a.40.2"' /></video></div>"
```
Comment (NicolasSiver) [inline data/html5video-rules.json:15]: Did you try your solution with relative urls?

Example: `./videos/presentation.mp4`

File: data/html5video-rules.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        },
+        {
+            "name": "OGV",
+            "displayName" : "OGV",
+            "icon": "",
+            "regex": "<a href="(https|http)?:?(.*\.ogv)".*>.*<\/a>",
```
Comment (NicolasSiver) [inline data/html5video-rules.json]: Could you add unit tests?
See: https://github.com/NicolasSiver/nodebb-plugin-ns-embed/blob/0ea86bfd58bdb4fea74b5a3bdfeb35c88eed52c2/test/regexRules.spec.js#L15-L18

**Review by NicolasSiver — CHANGES_REQUESTED** [review]

**Comment by Xuanwo** [comment]
Thanks for your review, I will do the job later.

**Pull request closed without merging**